src/Amqp/ChannelDispatcher.php

```diff
@@ -50,6 +50,8 @@ public function heartbeat(DeclarationManagerInterface $declarationManager)
             if (false === strpos($e->getMessage(), 'Failed to send heartbeat')) {
                 throw $e;
             }
+        } catch (ConnectionException $e) {
+            throw $e;
```

Comment:
Is something missing from here (otherwise, this is a no-op).

Reply:
These couple lines can still be removed, as they're simply re-throwing the same caught exception.